web/packages/teleport/src/Discover/Database/DeployService/AutoDeploy/SelectSecurityGroups.tsx

```diff
@@ -203,25 +203,22 @@ function withTips(
   securityGroups: SecurityGroup[],
   db: AwsRdsDatabase
 ): SecurityGroupWithRecommendation[] {
-  if (!db || !securityGroups) {
-    // return early without trying to add tips if there's no selected db or
-    // security groups given.
-    return securityGroups;
-  }
   const trustedGroups = getTrustedSecurityGroups(securityGroups, db);
-  return securityGroups.map(group => {
-    const isTrusted = trustedGroups.has(group.id);
-    const isOutboundAllowed = allowsOutbound(group);
-    return {
-      ...group,
-      tips: getTips(db, isTrusted, isOutboundAllowed),
-      // we recommend when either are true because security group rules are
-      // additive, meaning they can select multiple groups for a combined effect
-      // of satisfying the database inbound rules and the ECS task outbound
-      // rules.
-      recommended: isTrusted || isOutboundAllowed,
-    };
-  });
+  return (
+    securityGroups?.map(group => {
```

Comment:
i would just do the simple classic undefined/null check, at the begining of function. 
```ts
if (!securityGroups) {
    // return the expected type, otherwise we 
    // need to define return type as `SecurityGroupWithRecommendation[] | undefined`
    // and handle the `undefined` possibility at the call site
   return []
}
```

and change the function parameter types that it can be optional (if this is what we really expect)

```ts
function withTips(
  securityGroups?: SecurityGroup[],
  db: AwsRdsDatabase
): SecurityGroupWithRecommendation[] {
```

and then we dont have to `?` and `??` (which ?? btw wasn't necessary 😅 )

i know typescript doesn't help us out with this optional field atm, but it still gives more clues to waht to expect. i oftentimes have to go look at the original data type to see what is optional or not to help me be more intentional with my `?`